src/index.ts

```diff
@@ -66,6 +66,37 @@ export class Collection<K, V> extends Map<K, V> {
 		return super.clear();
 	}
 
+	/**
+	 * Gets an element if the key exists, otherwise sets and returns {@param getDefaultValue}'s return value.
+	 * @param {*} key - Key to get from/set to the collection.
+	 * @param {function} getDefaultValue - Function that returns the default value to be set and returned if the key doesn't exist.
+	 * @returns {*} The existing value if any, {@param getDefaultValue}'s return value otherwise.
+	 * @example <caption>Example use case: per-guild settings</caption>
+	 * // should be customized for what your bot needs
+	 * const defaultSettings = {
+	 *   welcomeChannelID: null,
+	 *   reactionRoleMessageID: null,
+	 *   adminRoleID: null,
+	 *   muteRoleID: null
+	 * }
+	 *
+	 * client.on('guildMemberAdd', member => {
+	 *   const getGuildSettings = client.settings.ensure(message.guild.id, () => defaultSettings);
+	 *
+	 *   // use the data
+	 *   if (guildSettings.welcomeChannelID) {
+	 *     const channel = client.channels.fetch(guildSettings.welcomeChannelID);
+	 *     channel.send(`Hello <@${member.id}>, welcome to ${member.guild.name}! Enjoy your stay!`);
+	 *   }
+	 * })
+	 */
```

Comment:
This example is too big.

It also uses a lot of discord.js stuff that is not really related to a generic collection library. Preferably we just remove this whole example. The docstring is explaining well enough what this method does.

Reply:
Upsert seems like it should be a separate method since they do different things. From what I understand, at least, upsert either inserts a value or updates it, and ensure is to make sure that there's an entry for that key *without* changing existing entries.